tensorboard/plugins/custom_scalar/README.md

```diff
@@ -0,0 +1,175 @@
+# The Custom Scalar Dashboard
+
+## The Problem this Dashboard Solves
```

Comment:
I'm not sure how committed we are to a particular logical structure for the README files, but personally as a relative newcomer I think I'd find this content a bit easier to digest if the first two sections here were swapped, so that first we get an explanation of what the dashboard does at a very high level - the "Overview" below - first, and then we delve into the problem that it solves.

Even though the "problem" part motivates the dashboard's existence and hence comes first in a causal sense, I think as a reader it's a little easier to get a first impression of what functionality the dashboard provides first, and then read about the motivation for it.  Especially since the latter may actually require more context.  E.g. in this case, IMO it's fairly straightforward to understand the Overview since it's just "custom line charts of scalar data" which I can understand without any prior TensorBoard exposure.  In contrast, the "problem solved" section requires more awareness of the limitations of the original scalar dashboard, the structure of tag names, the notion of the viewport, etc.  So it's helpful going into that to at least know exactly what this dashboard does do.

Just my $0.02 though.

Reply:
That makes sense to me. The problem part had also been lengthy, which had prevented the reader from understanding the main idea of the plugin til much later.

How about now? Overview is now first, and problem comes after. I also added a screenshot above the fold of the viewport.

Reply:
Looks good to me!